### PR TITLE
Translating ch10-01 in French.

### LINKED ISSUE #107: Translating ch10-01
I'm working on the French translation of this page.

### PR DESCRIPTION
**Pull request by Jimskapt**
Closes #107.

### CODE REVIEW

File: FRENCH/src/ch10-01-syntax.md [hunk 1]
```diff
@@ -0,0 +1,866 @@
...
+
+<!--
+When defining a function that uses generics, we place the generics in the
+signature of the function where we would usually specify the data types of the
+parameters and return value. Doing so makes our code more flexible and provides
+more functionality to callers of our function while preventing code duplication.
+-->
+
+Lorsque nous définissons une fonction en utilisant la généricité, nous utilisons
+des génériques dans la signature de la fonction où nous précisons habituellement
```
Comment (MartinHeywang) [inline FRENCH/src/ch10-01-syntax.md:35]:
```suggestion
des types génériques dans la signature de la fonction où nous précisons
habituellement
```

**Review by MartinHeywang — CHANGES_REQUESTED** [review]
Proofread ch10-01 !

Ignore my last review, I messed up.
And I used GitHub suggestions this time (as you can see)

File: FRENCH/src/ch10-01-syntax.md [hunk 2]
```diff
@@ -0,0 +1,866 @@
...
+Rust accomplishes this by performing monomorphization of the code that is using
+generics at compile time. *Monomorphization* is the process of turning generic
+code into specific code by filling in the concrete types that are used when
+compiled.
+-->
+
+Rust accomplit cela en pratiquant la monomorphisation à la compilation du code
+qui utilise les génériques. La *monomorphisation* est un processus pour
+transformer du code générique en code spécifique qui les remplace par des types
+concrets qui sont utilisés au moment de la compilation.
```
[the remark this answers is not in the document]
Comment (Jimskapt) [reply FRENCH/src/ch10-01-syntax.md:722]: Do you agree with this tiny change ?

```suggestion
Rust accomplit cela en pratiquant la monomorphisation à la compilation du code
qui utilise les génériques. La *monomorphisation* est un processus qui transforme
du code générique en code spécifique en définissant au moment de la compilation les
types concrets utilisés dans le code.
```

File: FRENCH/src/ch10-01-syntax.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jimskapt) [reply FRENCH/src/ch10-01-syntax.md:35]: Awaiting point of view of @joeljpresent about #270 😉

**Review by Jimskapt — COMMENTED** [review]
Merci ! 👍

**Review by joeljpresent — APPROVED** [review]

**Pull request merged**